### PR TITLE
Disputable apps: Implement Aragon app fees cashier interface

### PR DESCRIPTION
**Pull request by bingen**
- Add payment endpoint
- Rename methods to be “transaction specific”
- Add events

### CODE REVIEW

**Comment by bingen** [comment]
Mostly motivated by discussion [here](https://github.com/aragon/aragon-apps/issues/1186#issuecomment-652076269)

**Review by sohkai — APPROVED** [review]
LGTM, although I actually don't mind the old names :shrug:

**Comment by bingen** [comment]
The new names are because if we end up having a Subscripitons contract that implements also the Transactions interface, things like `getFees` would be too ambiguous.

**Comment by bingen** [comment]
Also, I’m thinking of removing the `Oracle` part from the interface name, as now it also has the payments endpoint.

**Review by facuspagnuolo — COMMENTED** [review]
Based on our discussion offline, I would:
- Rename the interface to `AragonAppFeesCashier`
- Rename the `(.*)(TransactionFee)(s)?` methods to `$1AppFee$3`
- Rename `uint256 _actionId` to a `bytes _data` instead

File: contracts/lib/arbitration/IAragonAppFeesCashier.sol [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+interface IAragonAppFeesCashier {
+    event AppFeeSet(bytes32 indexed appId, ERC20 token, uint256 amount);
+    event AppFeeUnset(bytes32 indexed appId);
+    event AppFeePaid(address indexed by, bytes32 appId, uint256 actionId);
+
+    function setAppFee(bytes32 _appId, ERC20 _token, uint256 _amount) external;
+    function setAppFees(bytes32[] _appIds, ERC20[] _tokens, uint256[] _amounts) external;
+    function unsetAppFee(bytes32 _appId) external;
+    function unsetAppFees(bytes32[] _appIds) external;
+    function payAppFees(bytes32 _appId, uint256 _actionId) external;
```
Comment (facuspagnuolo) [inline contracts/lib/arbitration/IAragonAppFeesCashier.sol]: Let use a `bytes data` here instead of `uint256 actionId`

**Review by facuspagnuolo — APPROVED** [review]
LGTM!

**Pull request merged**